microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/model/ProgrammingModelTest.java

```diff
@@ -0,0 +1,155 @@
+import org.jboss.eap.qe.microprofile.openapi.apps.routing.provider.services.InMemoryDistrictService;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.ConfigurationException;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientRelatedException;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+
+/**
+ * Test cases for MP OpenAPI programming model (migrated from RHOAR QE TS)
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class ProgrammingModelTest {
+
+    private final static String DEPLOYMENT_NAME = ProgrammingModelTest.class.getSimpleName();
+
+    private static String openApiUrl;
+    private static OnlineManagementClient onlineManagementClient;
+
+    @BeforeClass
+    public static void setup() throws ManagementClientRelatedException, ConfigurationException {
+        openApiUrl = OpenApiTestCaseHelper.composeDefaultOpenApiUrl();
+        //  MP OpenAPI up
+        onlineManagementClient = ManagementClientProvider.onlineStandalone();
+        if (!OpenApiManagementClientHelper.openapiSubsystemExists(onlineManagementClient)) {
+            OpenApiManagementClientHelper.enableOpenApi(onlineManagementClient);
+        }
+    }
+
+    @AfterClass
+    public static void tearDown() throws ManagementClientRelatedException, IOException {
+        //  MP OpenAPI down
+        try {
+            OpenApiManagementClientHelper.disableOpenApi(onlineManagementClient);
+        } finally {
+            onlineManagementClient.close();
+        }
+    }
+
+    @Deployment(testable = false)
+    public static WebArchive createCentralDeployment() {
+        WebArchive deployment = ShrinkWrap.create(
+                WebArchive.class,
+                String.format("%s.war", DEPLOYMENT_NAME))
+                .addClasses(RestApplication.class)
+                .addClasses(
+                        District.class,
+                        DistrictEntity.class,
+                        DistrictService.class,
+                        InMemoryDistrictService.class,
+                        DistrictsResource.class,
+                        RoutingServiceConstants.class)
+                .addClasses(OpenApiModelReader.class, OpenApiFilter.class)
+                .addAsResource(
+                        "META-INF/microprofile-config.properties")
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+
+        System.out.println("************************************************************");
```

Comment:
`System.out.printl(...);` should not be used and removed.

Reply:
Done.